### PR TITLE
Add new recipe for SFML

### PR DESCRIPTION
**Pull request by Gnimuc**

### CODE REVIEW

**Comment by Gnimuc** [comment]
The logs on Windows look weird. It complains about `/agent/_work/1/s/S/SFML/build/i686-w64-mingw32/hXxWmztE/destdir/bin/sfml-window-2.dll` at first, but then argues "Could not locate libsfml-window, sfml-window-2 inside ["/agent/_work/1/s/S/SFML/build/i686-w64-mingw32/hXxWmztE/destdir/bin".

```
[ Info: /agent/_work/1/s/S/SFML/build/i686-w64-mingw32/hXxWmztE/destdir/bin/sfml-window-2.dll locks us to cxx03
┌ Warning: /agent/_work/1/s/S/SFML/build/i686-w64-mingw32/hXxWmztE/destdir/bin/sfml-window-2.dll contains std::string values!  This causes incompatibilities across the GCC 4/5 version boundary.  To remedy this, you must build a tarball for both GCC 4 and GCC 5.  To do this, immediately after your `platforms` definition in your `build_tarballs.jl` file, add the line:
│ 
│     platforms = expand_cxxstring_abis(platforms)
└ @ BinaryBuilder.Auditor /depot/packages/BinaryBuilder/PtqUF/src/auditor/compiler_abi.jl:223
##[warning]/agent/_work/1/s/S/SFML/build/i686-w64-mingw32/hXxWmztE/destdir/bin/sfml-window-2.dll contains std::string values!  This causes incompatibilities across the GCC 4/5 version boundary.  To remedy this, you must build a tarball for both GCC 4 and GCC 5.  To do this, immediately after your `platforms` definition in your `build_tarballs.jl` file, add the line:

    platforms = expand_cxxstring_abis(platforms)
[ Info: Checking bin/sfml-window-2.dll with RPath list String[]
[ Info: Ignored system libraries OPENGL32.DLL, USER32.dll, WINMM.DLL, libstdc++-6.dll, GDI32.dll, msvcrt.dll, KERNEL32.dll, ADVAPI32.dll, libgcc_s_sjlj-1.dll
[ Info: Checking license file
[ Info: Found license file(s): license.md
[ Info: Found a valid dl path openal32.dll while looking for libsfml-window, sfml-window-2
[ Info: Found a valid dl path sfml-audio-2.dll while looking for libsfml-window, sfml-window-2
[ Info: Found a valid dl path sfml-graphics-2.dll while looking for libsfml-window, sfml-window-2
[ Info: Found a valid dl path sfml-network-2.dll while looking for libsfml-window, sfml-window-2
[ Info: Found a valid dl path sfml-system-2.dll while looking for libsfml-window, sfml-window-2
[ Info: Found a valid dl path sfml-window-2.dll while looking for libsfml-window, sfml-window-2
[ Info: Could not locate libsfml-window, sfml-window-2 inside ["/agent/_work/1/s/S/SFML/build/i686-w64-mingw32/hXxWmztE/destdir/bin"]
┌ Error: Built SFML but libsfml_window still unsatisfied:
└ @ BinaryBuilder /depot/packages/BinaryBuilder/PtqUF/src/AutoBuild.jl:759
##[error]Built SFML but libsfml_window still unsatisfied:

```

File: S/SFML/build_tarballs.jl [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+script = raw"""
+# build SFML
+cd ${WORKSPACE}/srcdir
+
+if [[ "${target}" == *linux* ]]; then
+
+cd SFML-2.5.1/
+mv ./include $WORKSPACE/destdir/
+mv ./lib $WORKSPACE/destdir/
+mv ./share $WORKSPACE/destdir/
```
Comment (giordano) [inline S/SFML/build_tarballs.jl:30]:
```suggestion
    cd SFML-2.5.1/
    mv ./include ./lib ./share ${prefix}/.
```
But why don't you build this from source?

Comment (Gnimuc) [reply S/SFML/build_tarballs.jl]: It depends on https://github.com/kcat/openal-soft which hasn't been built as a JLL package yet.

File: S/SFML/build_tarballs.jl [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+# Bash recipe for building across all platforms
+script = raw"""
+# build SFML
+cd ${WORKSPACE}/srcdir
+
+if [[ "${target}" == *linux* ]]; then
+
+cd SFML-2.5.1/
+mv ./include $WORKSPACE/destdir/
+mv ./lib64 $WORKSPACE/destdir/
```
Comment (Gnimuc) [inline S/SFML/build_tarballs.jl:29]: which env variable should I use here? is it `libdir`?

Comment (giordano) [inline S/SFML/build_tarballs.jl:29]: Please don't hardcode `$WORKSPACE/destdir/`: I don't think the documentation recommends using it anywhere (and if it does, it's an error), that path may change in the future, and it is even much longer to type than `${prefix}`
```suggestion
mv ./include ./lib64 ${prefix}
```
Also, [this question](https://github.com/JuliaPackaging/Yggdrasil/pull/1722#discussion_r498364581) still stands

File: S/SFML/build_tarballs.jl (hunk 1 above)
Comment (giordano) [reply S/SFML/build_tarballs.jl]: > which hasn't been built as a JLL package yet.

Any particular reason for not doing it?

Comment (Gnimuc) [reply S/SFML/build_tarballs.jl]: No. I was planning to upgrade CSFML to use JLLs firstly. I'm looking into the details now, but it looks like OpenAL depends on many backends as well. The SFML [doc](https://www.sfml-dev.org/tutorials/2.5/compile-with-cmake.php) just suggests using the binaries from Linux distributions, not sure how to configure those backends. I'll try it again tomorrow.

File: S/SFML/build_tarballs.jl [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+
+# Bash recipe for building across all platforms
+script = raw"""
+# build SFML
+cd ${WORKSPACE}/srcdir
+
+if [[ "${target}" == *linux* ]]; then
+
+cd SFML-2.5.1/
+mv ./include ./lib64 ${prefix}
```
Comment (giordano) [inline S/SFML/build_tarballs.jl:28]:
```suggestion
mkdir -p ${prefix}
mv ./include ./lib64 ${prefix}
```

Comment (giordano) [reply S/SFML/build_tarballs.jl:29]:
```suggestion
mv ./include ./lib ${prefix}
```

**Comment by giordano** [comment]
You miss quite some dependencies, including libopenal

File: S/SFML/build_tarballs.jl [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+
+
+if [[ "${target}" == *-linux-* ]]; then
+    # I know I'm doing something evil here...
+    apk add eudev-dev
+    cp /usr/include/libudev.h ${includedir}
+    cp /usr/include/udev.h ${includedir}
+    cp /usr/lib/libudev.a ${libdir}
+    cp /usr/lib/libudev.so ${libdir}
+fi
```
Comment (Gnimuc) [inline S/SFML/build_tarballs.jl:34]: OpenAL seems to work fine. Now the only problem is udev.

File: S/SFML/build_tarballs.jl [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+cd ${WORKSPACE}/srcdir
+
+cd SFML
+mkdir build && cd build
+
+CMAKE_FLAGS="-DCMAKE_INSTALL_PREFIX=${prefix} -DCMAKE_TOOLCHAIN_FILE=${CMAKE_TARGET_TOOLCHAIN} -DCMAKE_BUILD_TYPE=Release"
+
+
+if [[ "${target}" == *-linux-* ]]; then
+    # I know I'm doing something evil here...
```
Comment (giordano) [inline S/SFML/build_tarballs.jl:28]: It's not just evil, it's plainly wrong :wink:

Comment (Gnimuc) [reply S/SFML/build_tarballs.jl:28]: I know. Just wanna see whether it could fake BB2 or not. :innocent:

**Comment by Gnimuc** [comment]
I'd like to try again the old way that directly shipping pre-built binaries on Linux. This time we have all the runtime dependencies, hopefully, it can pass the auditing phase.

File: S/SFML/build_tarballs.jl [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+install_license ../license.md
+
+fi
+
+"""
+
+# These are the platforms we will build for by default, unless further
+# platforms are passed in on the command line
+platforms = [
+    Platform("x86_64", "linux"; libc=:glibc),
```
Comment (giordano) [inline S/SFML/build_tarballs.jl:73]:
```suggestion
    Platform("x86_64", "linux"; libc="glibc", cxxstring_abi="cxx11"),
```

File: S/SFML/build_tarballs.jl [hunk 7]
```diff
@@ -0,0 +1,102 @@
...
+    Platform("x86_64", "macos"),
+    Platform("i686", "windows"),
+    Platform("x86_64", "windows")
+]
+# platforms = expand_cxxstring_abis(platforms)
```
Comment (giordano) [inline S/SFML/build_tarballs.jl:78]:
```suggestion
platforms = expand_cxxstring_abis(platforms; skip=Sys.islinux)
```

**Comment by Gnimuc** [comment]
The failure seems unrelated. Is it ok to be merged? @giordano

**Pull request merged**